keygen

```diff
@@ -35,13 +35,17 @@ done
 command -v sed >/dev/null 2>&1 || { echo >&2 "ERROR: Please install the sed utility to generate key configuration."; exit 1; }
 
 # verify openssl is present and sufficiently recent (genpkey seems to require openssl 1.0+)
-command -v openssl >/dev/null 2>&1 || { echo >&2 "ERROR: Please install the openssl utility version 1.0.0 or newer to generate keys."; exit 1; }
-OPENSSL_VERSION_REGEX_MAJOR_BACKREF="(Open|Libre)SSL ([0-9]+).*"
+command -v openssl >/dev/null 2>&1 || { echo >&2 "ERROR: Please install the openssl utility version 1.1.1 or newer to generate keys."; exit 1; }
 OPENSSL_VERSION_STRING=$(openssl version)
+OPENSSL_VERSION_REGEX_MAJOR_BACKREF="(Open|Libre)SSL ([0-9]+).*"
 OPENSSL_VERSION_MAJOR=$(echo "$OPENSSL_VERSION_STRING" | sed -En "s/$OPENSSL_VERSION_REGEX_MAJOR_BACKREF/\2/p")
+OPENSSL_VERSION_REGEX_MINOR_BACKREF="(Open|Libre)SSL ${OPENSSL_VERSION_MAJOR}\.([0-9]+).*"
+OPENSSL_VERSION_MINOR=$(echo "$OPENSSL_VERSION_STRING" | sed -En "s/$OPENSSL_VERSION_REGEX_MINOR_BACKREF/\2/p")
+OPENSSL_VERSION_REGEX_PATCH_BACKREF="(Open|Libre)SSL ${OPENSSL_VERSION_MAJOR}\.${OPENSSL_VERSION_MINOR}\.([0-9]+).*"
+OPENSSL_VERSION_PATCH=$(echo "$OPENSSL_VERSION_STRING" | sed -En "s/$OPENSSL_VERSION_REGEX_PATCH_BACKREF/\2/p")
```

Comment:
Please let's be consistent with the syntax of bash variables. I suggest we use `${...}` everywhere.

Reply:
I'm fine with that. Added the brackets into all places in the script.